hazelcast/src/main/java/com/hazelcast/internal/ascii/HttpContentTypeParser.java

```diff
@@ -0,0 +1,143 @@
+     * It also returns null when charset parameter is malformed.
+     *
+     * @param contentTypeValue content type value
+     * @return charset if found and is a valid charset otherwise null
+     */
+    public static Charset parseCharset(String contentTypeValue) {
+        if (contentTypeValue == null) {
+            return null;
+        }
+        // parameter name tokens are case insensitive
+        contentTypeValue = contentTypeValue.toLowerCase(LOCALE);
+
+        // find the charset parameter token and remove it
+        int i = contentTypeValue.indexOf(CHARSET_PARAMETER_KEY_NAME);
+        if (i == -1 || i == 0) {
+            // content type cannot start with charset. there must be media type first
+            return null;
+        }
+        char charBeforeParamKey = contentTypeValue.charAt(i - 1);
+        if (charBeforeParamKey != PARAMETER_DELIMITER && charBeforeParamKey != ' ') {
+            return null;
+        }
+        String charset = contentTypeValue.substring(i + CHARSET_PARAMETER_KEY_NAME_LENGTH);
+
+        // remove possible trailing space or semicolons
+        i = charset.indexOf(' ');
+        if (i != -1) {
+            charset = charset.substring(0, i);
+        }
+        i = charset.indexOf(PARAMETER_DELIMITER);
+        if (i != -1) {
+            charset = charset.substring(0, i);
+        }
+
+        // parameters are allowed to be in double-quotes. let's remove them.
+        i = charset.indexOf('"');
+        if (i != -1) {
+            if (i != 0) {
+                return null;
+            }
+            if (charset.length() == 1) {
+                return null;
+            }
+            int i2 = charset.indexOf('"', 1);
+            if (i2 != charset.length() - 1) {
+                return null;
+            }
+            charset = charset.substring(i + 1, i2);
+        }
+
+        if (charset.isEmpty()) {
+            return null;
+        }
+        try {
+            return Charset.forName(charset);
+        } catch (IllegalCharsetNameException | UnsupportedCharsetException e) {
+            return null;
+        }
+    }
+
+    /**
+     * TODO
+     *
+     * @param contentTypeValue
+     * @return
+     */
+    public static String parseMediaType(String contentTypeValue) {
+        if (contentTypeValue == null) {
+            return null;
+        }
+        contentTypeValue = contentTypeValue.trim().toLowerCase(LOCALE);
+
+        int i = contentTypeValue.indexOf(SUBTYPE_DELIMITER);
+        if (i == -1) {
+            return null;
+        }
+
+        // remove trailing semicolon
+        i = contentTypeValue.indexOf(PARAMETER_DELIMITER);
+        if (i != -1) {
+            contentTypeValue = contentTypeValue.substring(0, i);
+        }
+
+        // remove trailing space
+        i = contentTypeValue.indexOf(' ');
+        if (i != -1) {
+            contentTypeValue = contentTypeValue.substring(0, i);
+        }
```

Comment:
What about moving the `trim`+`lowerCase` from the method beginning to here? So we wouldn't need this piece.